python/cudf_polars/cudf_polars/dsl/utils/aggregations.py

```diff
@@ -219,7 +219,49 @@ def decompose_single_agg(
                 expr.Col(agg.dtype, name)
             )
     if isinstance(agg, expr.Ternary):
-        raise NotImplementedError("Ternary inside groupby")
+        when, then, otherwise = agg.children
+
+        when_aggs, when_post = decompose_single_agg(
+            expr.NamedExpr(next(name_generator), when),
+            name_generator,
+            is_top=False,
+            context=context,
+        )
+        then_aggs, then_post = decompose_single_agg(
+            expr.NamedExpr(next(name_generator), then),
+            name_generator,
+            is_top=False,
+            context=context,
+        )
+        otherwise_aggs, otherwise_post = decompose_single_agg(
+            expr.NamedExpr(next(name_generator), otherwise),
+            name_generator,
+            is_top=False,
+            context=context,
+        )
+
+        when_has = any(h for _, h in when_aggs)
+        then_has = any(h for _, h in then_aggs)
+        otherwise_has = any(h for _, h in otherwise_aggs)
+
+        if is_top and not (when_has or then_has or otherwise_has):
+            raise NotImplementedError(
+                "Broadcasted ternary with list output in groupby is not supported"
+            )
+
+        for post, has in (
+            (when_post, when_has),
+            (then_post, then_has),
+            (otherwise_post, otherwise_has),
+        ):
+            if is_top and not has and not isinstance(post.value, expr.Literal):
+                raise NotImplementedError(
+                    "Broadcasting aggregated expressions in groupby/rolling"
```

Comment:
Is there a reason this says "groupby/rolling" while the [one above](https://github.com/rapidsai/cudf/pull/19242/files#diff-e8daefeadd44cc548a51e8e043125160a4452c26f36a3ef8634e809f01569749R249) says just "groupby"?

Reply:
They should all probably mention groupby/rolling since both operations hit this code path. I can update them all in a follow up because its also possible that polars has differing behavior for aggs in groupby vs a rolling context. And I would need to check each one first before updating the error message.